src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -23,14 +23,12 @@ import { SimpleSavedObject } from './simple_saved_object';
 import type { ResolvedSimpleSavedObject } from './types';
 import { HttpFetchOptions, HttpSetup } from '../http';
 
-type PromiseType<T extends Promise<any>> = T extends Promise<infer U> ? U : never;
```

Comment:
`Awaited` is now available globally, provided by the TypeScript lib and serves the same function as `PromiseType`.